hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -107,6 +122,15 @@ public String getPath() {
     return path;
   }
 
+  private String getInputStreamID() {
+    return StringUtils.right(UUID.randomUUID().toString(), STREAM_ID_LEN);
+  }
+
+  public void registerListener(Listener listener1) {
```

Comment:
Same as before. Methods used by test need to be private and annotated with VisibleForTesting

Reply:
Can not make it private since the test classes using it are in a different package. Have moved to the end of file where other VisibleForTesting methods are written (which are also public due to the same reason)